### DISCUSSION TITLE
Swagger-Ui file disappears after deploy

### DISCUSSION THREAD

**Discussion by Kimishu** [juhaku/utoipa#1261] [Q&A] [answered]
Hi again!
I've reached some problem when deploying my app on a server. Let's figure it out more clearly:
1) When I tested my app on local machine, despite the different OS, all works fine. My app works clearly on `Windows/Linux(WSL)`, so.. I can easily navigate to `https://localhost:8000/swagger-ui`.
2) After deploy to a server, my app exists in a docker container. All routes works fine and I even can get an openapi.json by navigating to `https://ip:port/api-docs/openapi.json`. All works fine except `swagger-ui` route

I have no idea where problem source is (something goes wrong with deploy or something deleted after build, maybe..?). If someone has any ideas - please, share it with me!
Thanks in advance!

UPD: I found the problem. When app was builded, deploy mechanism get's only binary file `main` and don't take, as I understand, utoipa files. So, now I should somehow detect what files I need and which not.

**Comment by juhaku** [comment, marked as answer]
So you cannot see Swagger UI after you have build the docker container? `utoipa-swagger-ui` uses `rust-embed` to embed the Swagger UI files into the binary. The `rust-embed` does not embed files on debug build to the binary. Instead you need to either build the project with `--release` flag or enabled `debug-embed` feature. https://github.com/juhaku/utoipa?tab=readme-ov-file#swagger-ui-returns-404-notfound-from-built-binary

Here is an example with Docker: https://github.com/juhaku/utoipa/tree/master/examples/axum-utoipa-nesting-vendored

Here you can find the `utoipa-swagger-ui` feature flags: https://github.com/juhaku/utoipa/tree/master/utoipa-swagger-ui#crate-features
